geo/src/main/java/com/tomtom/speedtools/geometry/GeoPolyLine.java

```diff
@@ -50,26 +50,38 @@ public GeoPolyLine(@Nonnull final List<GeoPoint> points) {
         assert points != null;
         assert points.size() >= 2;
 
-        // Find the first elevation.
-        double elevationMeters = Double.NaN;
-        for (final GeoPoint point : points) {
-            if (point.getElevationMeters() != null) {
-                elevationMeters = point.getElevationMetersOrNaN();
-                break;
-            }
-        }
+        // Check the need to correct the elevation in a list of points.
+        final List<GeoPoint> elevatedPoints;
+        final boolean noneElevated = points.stream().allMatch(x -> (x.getElevationMeters() != null));
+        final boolean allElevated = points.stream().allMatch(x -> (x.getElevationMeters() == null));
```

Comment:
I don't know how expensive these operations are, but a nice optimization could be to only check for `noneElevated` if `allElevated == false`.

Reply:
By using noneMatch twice, they should be cheap.